pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator/indexScript.js

```diff
@@ -0,0 +1,34 @@
+Behaviour.specify(".directive-generator-button-reference-holder", 'prototype', 0, function (e) {
+    var url = e.getAttribute('data-fullurl');
+    var id = e.getAttribute('data-id');
+    var button = document.getElementById('prototypeButton_'+id);
+
+    button.onclick = function(el) {
+       handlePrototype(url,id);
+       return false;
+    }
+});
+function handlePrototype(url,id) {
+    buildFormTree(document.forms.config);
+    // TODO JSON.stringify fails in some circumstances: https://gist.github.com/jglick/70ec4b15c1f628fdf2e9 due to Array.prototype.toJSON
+    // TODO simplify when Prototype.js is removed
```

Comment:
likely needs to be revisited in another PR

Reply:
Can be done. Will work on it after this one